feat(ui): add error boundaries

ctjhoa opened the pull request:
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1039

RemiBonnet commented:
[![Qovery Preview](https://www.qovery.com/images/logo-white.svg)](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> There are two hard problems in computer science: naming things, cache invalidation and off-by-1 errors

RemiBonnet commented on libs/shared/ui/src/lib/components/error-boundary/error-boundary.tsx:
```diff
@@ -0,0 +1,42 @@
+import { type PropsWithChildren } from 'react'
+import { ErrorBoundary as ErrorBoundaryRaw } from 'react-error-boundary'
+import { type FallbackProps } from 'react-error-boundary'
+import { useLocation } from 'react-router-dom'
```
Duplicate import for `react-error-boundary`

ctjhoa replied:
Fixed https://github.com/Qovery/console/pull/1403/commits/b57dc584729cbe16937b5db1f215935ff9bed56a

RemiBonnet commented on libs/shared/ui/src/lib/components/error-boundary/error-boundary.tsx:
```diff
@@ -0,0 +1,42 @@
+
+function ErrorFallback({ error, resetErrorBoundary }: FallbackProps) {
+  return (
+    <Section className="flex-grow items-center justify-center rounded-sm bg-white">
```
I think we need to remove the `justify-center` and have an static margin top to avoid this kind of view

<img width="1708" alt="Screenshot 2024-06-05 at 16 25 14" src="https://github.com/Qovery/console/assets/533928/fde73cd4-8adb-4f0b-bef4-684c59b4ecc8">

RemiBonnet reviewed (changes requested):
I think it missing an `ErrorBoundary` on the `main.tsx` level to avoid error from container or parent page

```      <BrowserRouter>
        <ErrorBoundary>
          <TooltipProvider>
            <ModalProvider>
              <App />
              <ToastBehavior />
            </ModalProvider>
          </TooltipProvider>
        </ErrorBoundary>
      </BrowserRouter>```

I tested other several cases and look good for me except some comment I added on my review

ctjhoa commented:
> I think it missing an `ErrorBoundary` on the `main.tsx` level to avoid error from container or parent page
> 
> ```
>         <ErrorBoundary>
>           <TooltipProvider>
>             <ModalProvider>
>               <App />
>               <ToastBehavior />
>             </ModalProvider>
>           </TooltipProvider>
>         </ErrorBoundary>
>       </BrowserRouter>```
> 
> I tested other several cases and look good for me except some comment I added on my review
> ```

Fixed 
https://github.com/Qovery/console/pull/1403/commits/62f362e1d6f9108b2edb2af83adf6fd84d4cad68

RemiBonnet commented:
:tada: This PR is included in version 1.130.0 :tada:

The release is available on [GitHub release](https://github.com/Qovery/console/releases/tag/v1.130.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
